documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,267 @@
+            "pattern": "kafka.server<type=(.+), name=(.+)PerSec\\w*, topic=(.+)><>Count",
+            "name": "kafka_server_$1_$2_total",
+            "labels":
+            {
+              "topic": "$3"
+            }
+          }
+      ]
+    }
+  zookeeper-metrics-config: |-
+    {
+      "lowercaseOutputName": true
+    }
+----
+
+==== Storage
+
+Kafka needs to save information like data logs (so received messages), consumer offsets and so on in some way; the same is true for 
+Zookeeper about znodes children and values.
+The Strimzi project allows to save such a data in an "ephemeral" way (using `emptyDir`) or in a "persistent-claim" way using persistent 
+volumes.
+It's possible to provide the storage configuration in the related ConfigMap using a JSON string as value for the 
+`kafka-storage` and `zookeeper-storage` fields.
+
+The JSON representation has a mandatory `type` field for specifying the type of storage to use ("ephemeral" or "persistent-claim").
+In case of "persistent-claim" type the following fields can be provided as well :
+
+* `size`: defines the size of the persistent volume claim (i.e 1Gi)
+* `class` : the Kubernetes/OpenShift storage class to use for dynamic volume allocation
+* `selector`: allows to select a specific persistent volume to use. It contains a `matchLabels` field which defines an 
+inner JSON object with key:value representing labels for selecting such a volume.
+* `delete-claim`: specifies if the persistent volume claim has to be deleted when the cluster is un-deployed. Default is `false`
+
+The "ephemeral" storage is really simple to configure and the related JSON string is something like this.
+
+.Ephemeral storage JSON
+[source,json]
+----
+{ "type": "ephemeral" }
+
+----
+
+The "persistent" storage has some more parameters but other than `type`, the `size` is the only mandatory one.
+
+.Persistent storage JSON with 1Gi as size
+[source,json]
+----
+{ "type": "persistent-claim", "size": "1Gi" }
+----
+
+A more complex configuration could use a storage class in the following way.
+
+.Persistent storage JSON using "storage class"
+[source,json]
+----
+{
+  "type": "persistent-claim",
+  "size": "1Gi",
+  "class": "my-storage-class"
+}
+----
+
+Finally, a selector can be used in order to select a specific labeled persistent volume which provides some needed features (i.e. an SSD)
+
+.Persistent storage JSON with "match labels" selector
+[source,json]
+----
+{
+  "type": "persistent-claim",
+  "size": "1Gi",
+  "selector":
+  {
+    "matchLabels":
+    {
+      "hdd-type": "ssd"
+    }
+  },
+  "delete-claim": "true"
+}
+----
+
+==== Metrics
+
+Because the Strimzi project uses the [JMX exporter](https://github.com/prometheus/jmx_exporter) in order to expose metrics 
+on each node, the JSON string used for metrics configuration in the cluster ConfigMap reflects the related JMX exporter 
+configuration file. For this reason, you can find more information on how to use it in the corresponding GitHub repo.
```

Comment:
This should really include / link to [METRICS.md](https://github.com/strimzi/strimzi/blob/master/metrics/METRICS.md)